sources/Application/Views/BaseClasses/UITextField.cpp

```diff
@@ -70,9 +92,54 @@ void UITextField::ProcessArrow(unsigned short mask) {
   case EPBM_RIGHT:
     if (currentChar_ < len - 1) {
       currentChar_++;
+    } else if (currentChar_ < 39) {
+      currentChar_++;
+      name[currentChar_] = 'A';
+      name[currentChar_ + 1] = '\0';
+      len++;
+      src_->SetString(name, true);
+      SetChanged();
+      NotifyObservers((I_ObservableData *)fourcc_);
     }
     break;
   };
 };
 
 etl::string<40> UITextField::GetString() { return src_->GetString(); };
+
+char getNext(char c, bool reverse) {
+  // Valid characters in order
+  const char validChars[] =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._";
```

Comment:
yes I think limiting to uppercase would make sense and will need to update random name gen for that too, so maybe do as a followup PR?

Reply:
actually this will likely be used in future for (re)naming samples and instrument settings etc, so maybe we keep lowercase